### ISSUE TITLE
VGG16 and VGG16  tests failing in evaluation 

### ISSUE THREAD

**Issue by gits00** [iml130/nncg#3]
Ran tests using the code "applications/tests.py". 

Looks like VGG16_test()  test is failing in evaluation.

.....    
######################################################################
            strides passed
######################################################################    
    

2021-08-30 15:52:07.957106: W tensorflow/core/framework/cpu_allocator_impl.cc:80] Allocation of 411041792 exceeds 10% of free system memory.
2021-08-30 15:52:08.068794: W tensorflow/core/framework/cpu_allocator_impl.cc:80] Allocation of 411041792 exceeds 10% of free system memory.
2021-08-30 15:52:08.141630: W tensorflow/core/framework/cpu_allocator_impl.cc:80] Allocation of 411041792 exceeds 10% of free system memory.
Finished |████████████████████████████████████████| 100.0% 7.1% ---| 0.0% 
Check of variable x_13 for layer block3_conv1.
Largest error 2.3889937210419543e-07 at (31, 11, 94) (447326).
Values: 0.1227651908993721 and 0.122764952

VGG19_test() as well - 

....
....
....
2021-08-30 16:07:31.739931: W tensorflow/core/framework/cpu_allocator_impl.cc:80] Allocation of 411041792 exceeds 10% of free system memory.
2021-08-30 16:07:31.860670: W tensorflow/core/framework/cpu_allocator_impl.cc:80] Allocation of 411041792 exceeds 10% of free system memory.
2021-08-30 16:07:31.937269: W tensorflow/core/framework/cpu_allocator_impl.cc:80] Allocation of 411041792 exceeds 10% of free system memory.
Finished |████████████████████████████████████████| 100.0% 7.1% ---| 0.0% 
Check of variable x_15 for layer block3_conv2.
Largest error 2.2351569518852532e-07 at (9, 29, 124) (136572).
Values: 0.06643339991569519 and 0.0664331764

The quantized ones with real images are also failing since probably they use the same generated C code. 
I don't get a print_success if I comment out  VGG16_test() and VGG19_test() and let run VGG16_quantized_test(db),       VGG19_quantized_test(db)

**Comment by OliverUrbann** [comment]
There is always a difference between the Keras results and the NNCG results as the calculation is surely different and by rounding numbers etc. a small error can occur. Here, 10**-7 is in the range of the resolution of float, so that looks ok. An actuall error would be if the classification leads to different results, but that does not seem to be the case here. However, in the quantized tests also differences in classification are allowed as quantized networks are known to lose accuracy.
